supplier_api_tester/supplier_api_tester/v2/utils/availability.py

```diff
@@ -276,17 +258,39 @@ def not_allowed_method(api_url, api_key, product_id, endpoint, version=2):
     return TestResult()
 
 
-# def test_error_for_non_timeslot_product(api_url, api_key, product_id, endpoint, version=2):
-#     '''Testing timeslot availability for non timeslot product'''
-#     tomorrow = get_tomorrow()
-#     raw_response, response = client(f'{api_url}/v{version}/products/{product_id}/{endpoint}', api_key, {
-#         'start': tomorrow.isoformat(),
-#         'end': tomorrow.isoformat(),
-#     })
-#     api_error = get_api_error(raw_response, response)
-#     expected_error = ApiError(
-#         error_code=1002,
-#         error='Timeslot product expected',
-#         message=f'Requested timeslot availability for non timeslot product ID {product_id}',
-#     )
-#     return check_api_error(raw_response, api_error, expected_error)
+def product_with_provides_pricing(api_url, api_key, product_id, endpoint, version=2):
+    """Tests that the availability response for a product whose provides_pricing attribute is True contains the `price`
+    attribute for every variant id."""
+    catalog_response: Response
+    products: List[Product]
+    catalog_response, products = get_catalog(api_url, api_key, version)
+
+    if not product_provides_pricing(product_id, products):
+        return TestResult(
+            status=1,
+            message=(
+                "Skipping the test because the product does not provide pricing."
+            )
+        )
+
+    # retrieve the availability of the product
+    days: List[DailyVariants]
+    tomorrow = get_tomorrow()
```

Comment:
**nitpick (non-blocking/typing):** Will be nice to keep a consistent typing standard.